packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -947,6 +952,36 @@ function updateClassComponent(
   renderLanes: Lanes,
 ) {
   if (__DEV__) {
+    // This is used by DevTools to force a boundary to error.
+    switch (shouldError(workInProgress)) {
+      case false: {
+        const instance = workInProgress.stateNode;
+        const ctor = workInProgress.type;
+        const tempInstance = new ctor(
+          workInProgress.memoizedProps,
+          instance.context,
+        );
```

Comment:
Nit This was my suggested approach 😆 but it's a little hacky so maybe we can add a TODO comment above it in case we can think of a better approach later?
```suggestion
        // TODO This way of resetting the error boundary state is a hack.
        // Is there a better way to do this?
        const tempInstance = new ctor(
          workInProgress.memoizedProps,
          instance.context,
        );
```
(Since this is a change to `ReactFiberBeginWork.new.js` you'll also need to run `yarn replace-fork` to sync the change to `ReactFiberBeginWork.old.js`.)